teleport.lua

```diff
@@ -171,23 +171,29 @@ local function is_protected_beacon(pos, player)
 end
 
 -- digilines optional support
+local DEFAULT_CHANNEL = "telemosaic"
+
 local telemosaic_digiline_switching
 local telemosaic_digiline_base
 
 local on_digiline_receive = function(pos, _, channel, msg)
-	local setchan = minetest.get_meta(pos):get_string("channel")
+	local meta = minetest.get_meta(pos)
+	local setchan = meta:contains("channel") and meta:get_string("channel") or DEFAULT_CHANNEL
 	if channel ~= setchan then
 		return false
 	end
-	if type(msg) ~= "string" then
+	if type(msg) == "string" then
+		local a = string.split(msg, " ")
+		msg = { cmd = a[1], channel = a[2] }
+	end
+	if type(msg) ~= "table" then
 		return false
 	end
-	local cmd = string.split(msg, " ")
-	if cmd[1] == "channel" then
-		if #cmd > 1 then
-			minetest.get_meta(pos):set_string("channel", cmd[2])
+	if msg.cmd == "setchannel" then
+		if msg.channel ~= nil then
+			meta:set_string("channel", msg.channel)
 		else
-			minetest.get_meta(pos):set_string("channel", "")
+			meta:set_string("channel", DEFAULT_CHANNEL)
```

Comment:
I'd say remove this completely, maybe better to do nothing if incoming message does not contain valid channel.

Reply:
Right. Done.